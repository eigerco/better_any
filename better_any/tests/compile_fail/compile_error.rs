use better_any::nightly::downcast_any;
use std::any::Any;

fn testlt<'a, 'b>(any: &'a dyn Any) -> &'b i32 {
    downcast_any(any).unwrap()
}

fn main() {}
