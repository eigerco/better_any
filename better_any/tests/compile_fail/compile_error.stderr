error: lifetime may not live long enough
 --> $DIR/compile_error.rs:5:5
  |
4 | fn testlt<'a, 'b>(any: &'a dyn Any) -> &'b i32 {
  |           --  -- lifetime `'b` defined here
  |           |
  |           lifetime `'a` defined here
5 |     downcast_any(any).unwrap()
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ function was supposed to return data with lifetime `'b` but it is returning data with lifetime `'a`
  |
  = help: consider adding the following bound: `'a: 'b`
